torch/fx/experimental/Partitioner.py

```diff
@@ -126,19 +220,39 @@ def __init__(self) -> None:
         self.partitions: List[Partition] = []
         self.devices: List[Dict[str, Union[str, float]]] = []
         self.node_to_partitions: Dict[Node, List[int]] = {}
+        self.partition_to_used_mem: Dict[Partition, int] = {}
+        self.orphans: List[Node] = []
+        self.args_mapping: Dict[int, List[int]] = []
+        self.users_mapping: Dict[int, List[int]] = []
 
     def partition_graph(self, fx_module: GraphModule, devices: List[Dict[str, Union[str, float]]]) -> DAG:
         """
-        Given the fx_module and devices, find the partitions, do the partitions,
+        Given the fx_module and devices [{"name": "dev_0", "available_mem": float('inf')}],
+        find the partitions, and then do the partitions,
         and then return a dictionary representing the DAG of partitions
```

Comment:
It would help for the docstring to specify what the behavior actually is here. In particular, somethings to call out might include:

* What happens when the Module fits on one device? (single partition)
* What happens when module does not fit on one device but fits split across multiple devices? (size-based partition)
* What happens if memory requirements exceed the total memory across all devices?
* When would number of partitions exceed the number of available devices
* Are there situations where (total mem of devices) > (memory requirement of Module), but a partition cannot happen (e.g. fragmentation)?